csharp/src/Apache.Arrow/Extensions/StreamExtensions.netcoreapp2.1.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+
+namespace Apache.Arrow
+{
+    // Helpers to read from Stream to Memory<byte> on netcoreapp
+    internal static class StreamExtensions
+    {
+        public static int Read(this Stream stream, Memory<byte> buffer)
```

Comment:
Why is this necessary?  Why not just call it with buffer.Span at the call site?

Reply:
I see. You've added this as an abstraction to enable fishing out the array when the span-based overload doesn't exist on netstandard. Ok.

Reply:
Is there are better way to do this? I didn't find one.

Reply:
Given the desire to also compile for netstandard, this seems reasonable.